component/kafka/group/option.go

```diff
@@ -26,6 +29,52 @@ func FailureStrategy(fs kafka.FailStrategy) OptionFunc {
 	}
 }
 
+// CheckTopic will attempt to:
+//	1. connect to the broker,
+//  2. retrieve the existing topics in the broker
+//  3. if auto-topic creation is disabled, check whether the configured topics exist in the broker
+// If any of the checks above fail the component will exit before starting to consume messages
+func CheckTopic() OptionFunc {
+	return func(c *Component) error {
+		c.topicChecker = func(ctx context.Context, c *Component) error {
+			client, err := sarama.NewClient(c.brokers, c.saramaConfig)
+			defer func() {
+				if client != nil {
+					_ = client.Close()
+				}
+			}()
+			// without client no connection can be established
+			if err != nil {
+				return err
```

Comment:
Perhaps we should wrap this error to inform it was in the context of checking the topics?

Reply:
I think that makes sense, would be easier for testing purpose as well...

Reply:
Wrap here makes sense, I could not identify any Patron-error, how would you suggest me to wrap the error? 

1. Create a local error type that can be checked for outside by type
2. fmt.Errorf ....